File: C++/28.组合实体模式/CompositeEntityPatternDemo.cpp
/**
 * @file CompositeEntityPatternDemo.cpp
 * @author {Layne} ({shu_huanglei@163.com})
 * @brief
 * @version 0.1
 * @date 2019-12-04
 *
 * @copyright Copyright (c) 2019
 *
 */

#include "Client.h"

int main(int argc, char const *argv[]) {
    std::shared_ptr<Client> client = std::make_shared<Client>();
    client->SetData("Test", "Data");
    client->PrintData();
    client->SetData("Second Test", "Data1");
    client->PrintData();
    return 0;
}

File: Objective-C/28.CompositeEntityPattern/CompositeEntityPatternDemo.m
//
//  CompositeEntityPatternDemo.m
//  28.CompositeEntityPattern
//
//  Created by Layne on 2020/11/21.
//

#import "Client.h"

int main(int argc, const char* argv[]) {
    @autoreleasepool {
        Client* client = [[Client alloc] init];
        [client setData1:@"Test" Data2:@"Data"];
        [client printData];
        [client setData1:@"Second Test" Data2:@"Data1"];
        [client printData];
    }
    return 0;
}
